_posts/2018-10-20-reducers-exploring-state-management-in-react.md

```diff
@@ -0,0 +1,125 @@
+---
+layout: post
+title: Reducers&#58; Exploring State Management in React (Part 2)
+date: 2018-10-20 09:00:00 -04:00
+categories: post
+desc: Let's explore how to extract React state transformations to better isolate and understand the way our component state changes.
+permalink: /:categories/:year/:month/:day/:title/
+---
+
+In a previous [post](https://www.mitchellhanberg.com/post/2018/07/25/exploring-state-management-in-react-container-components/), we demonstrated how the Container/Presenter pattern is a solid approach to managing your React state. This time we are going to look into using Reducer functions as the method to managing change in state of your components.
+
+## Reduce
+
+Reduce (also known as a [fold](https://en.wikipedia.org/wiki/Fold_%28higher-order_function%29)) is a functional programming concept that deals with the transformation of data structures using recursion and higher order functions. If you have used either the `Array.prototype.reduce` or `Array.prototype.map` functions, you already have experience with this technique.
```

Comment:
💡 Would linking `reduce` and `map` to MDN (e.g. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) help?